statsd/aggregator.go

```diff
@@ -124,22 +114,20 @@ func (a *aggregator) flush() {
 	}
 }
 
-func (a *aggregator) flushTelemetryMetrics() *aggregatorMetrics {
+func (a *aggregator) flushTelemetryMetrics(t *Telemetry) {
 	if a == nil {
-		return nil
+		// aggregation is disabled
+		return
 	}
 
-	am := &aggregatorMetrics{
-		nbContextGauge:        atomic.SwapInt32(&a.nbContextGauge, 0),
-		nbContextCount:        atomic.SwapInt32(&a.nbContextCount, 0),
-		nbContextSet:          atomic.SwapInt32(&a.nbContextSet, 0),
-		nbContextHistogram:    a.histograms.resetAndGetNbContext(),
-		nbContextDistribution: a.distributions.resetAndGetNbContext(),
-		nbContextTiming:       a.timings.resetAndGetNbContext(),
-	}
+	t.aggEnabled = true
```

Comment:
I agree, not sure what we could do to improve it. WDYT ?

Reply:
I see how this plays well into some of the lock-less stuff you're doing, like the nil-pointer receiver checks.... Given that aggregation cannot be changed on the fly if I'm not mistaken, you could also set that up at initialization, but I guess this is fine otherwise.